src/NuGet.Core/NuGet.Packaging/PackageExtraction/StreamExtensions.cs

```diff
@@ -27,12 +34,24 @@ public static string CopyToFile(this Stream inputStream, string fileFullPath)
                 return fileFullPath;
             }
 
+            // For files of a certain size, we can do some Cleverness and mmap them instead
+            // of writing directly to disk. This can increase performance by a LOT, at the
+            // cost of files taking longer to get to disk -- but that's fine!
```

Comment:
I'm not sure I fully understand this statement, especially the "but that's fine!" part.  Perhaps it can be reworded and that phrase elaborated on.

Is there any scenario where files taking longer to get to disk can degrade performance?